### PR TITLE
⚠️ CONFLICT! Lineage pull request for: skeleton

### PR DESCRIPTION
**Pull request by cisagovbot**
# Lineage Pull Request: CONFLICT #

[Lineage] has created this pull request to incorporate new changes found in an
upstream repository:

Upstream repository: [`https://github.com/cisagov/skeleton-tf-module.git`](https://github.com/cisagov/skeleton-tf-module.git)
Remote branch: `HEAD`

Check the changes in this pull request to ensure they won't cause issues with
your project.

The `lineage/skeleton` branch has **one or more unresolved merge conflicts**
that you must resolve before merging this pull request!

## How to resolve the conflicts ##

1. Take ownership of this pull request by removing any other assignees.

1. Clone the repository locally, and reapply the merge:

    ```console
    git clone git@github.com:cisagov/cool-accounts.git cool-accounts
    cd cool-accounts
    git remote add skeleton https://github.com/cisagov/skeleton-tf-module.git
    git remote set-url --push skeleton no_push
    git switch develop
    git checkout -b lineage/skeleton --track origin/develop
    git pull skeleton HEAD
    git status
    ```

1. Review the changes displayed by the `status` command.  Fix any conflicts and
   possibly incorrect auto-merges.

1. After resolving each of the conflicts, `add` your changes to the
   branch, `commit`, and `push` your changes:

    ```console
    git add .github/CODEOWNERS .gitignore README.md examples/basic_usage/README.md examples/basic_usage/main.tf examples/basic_usage/outputs.tf examples/basic_usage/variables.tf main.tf outputs.tf variables.tf 
    git commit
    git push --force --set-upstream origin lineage/skeleton
    ```

    Note that you may _append_ to the default merge commit message
    that git creates for you, but *please do not delete the existing
    content*.  It provides useful information about the merge that is
    being performed.

1. Wait for all the automated tests to pass.

1. Check the "Everything is cool" checkbox below:

    - [x] ✌️ The conflicts in this pull request have been resolved.

1. Mark this draft pull request "Ready for review".

------------

**Note:** *You are seeing this because one of this repository's maintainers has
configured [Lineage] to open pull requests.*

For more information:

🛠 [Lineage] configurations for this project are stored in `.github/lineage.yml`

📚 [Read more about Lineage][Lineage]

[//]: # (lineage:metadata:{&quot;slayed&quot;:true})
[Lineage]: https://github.com/cisagov/action-lineage/ "Lineage GitHub Action"

### CODE REVIEW

**Review by jsf9k — APPROVED** [review]

**Review by dav3r — APPROVED** [review]
Thanks for slaying this beast! 🐲

**Review by mcdonnnj — APPROVED** [review]
I just had some suggestions to get consistent language for the `tags` variable and a period for another variable description. The changes would of course have to be made in the respective variable descriptions.

File: users/README.md [hunk 1]
```diff
@@ -65,41 +65,64 @@ future changes by simply running `terraform apply
...
+| self\_managed\_creds\_with\_mfa\_policy\_name | The name to assign the IAM policy that allows users to administer their own user accounts, requiring multi-factor authentication (MFA). | `string` | `"SelfManagedCredsWithMFA"` | no |
+| self\_managed\_creds\_without\_mfa\_policy\_description | The description to associate with the IAM policy that allows users to administer their own user accounts, without requiring multi-factor authentication (MFA). | `string` | `"Allows sufficient access for users to administer their own user accounts, without requiring multi-factor authentication (MFA)."` | no |
+| self\_managed\_creds\_without\_mfa\_policy\_name | The name to assign the IAM policy that allows users to administer their own user accounts, without requiring multi-factor authentication (MFA). | `string` | `"SelfManagedCredsWithoutMFA"` | no |
 | tags | Tags to apply to all AWS resources provisioned. | `map(string)` | `{}` | no |
```
[the remark this answers is not in the document]
Comment (jsf9k) [reply users/README.md]: See commit 87991b5.

File: terraform/README.md [hunk 2]
```diff
@@ -86,51 +86,86 @@ future changes by simply running `terraform apply
...
+| read\_terraform\_state\_role\_name | The name to assign the IAM role (as well as the corresponding policy) that allows read-only access to the S3 bucket where Terraform state is stored. | `string` | `"ReadTerraformState"` | no |
+| state\_bucket\_name | The name to use for the S3 bucket that will store the Terraform state. | `string` | n/a | yes |
+| state\_table\_name | The name to use for the DynamoDB table that will be used for Terraform state locking. | `string` | `"terraform-state-lock"` | no |
+| state\_table\_read\_capacity | The number of read units for the DynamoDB table that will be used for Terraform state locking. | `number` | `20` | no |
+| state\_table\_write\_capacity | The number of write units for the DynamoDB table that will be used for Terraform state locking. | `number` | `20` | no |
 | tags | Tags to apply to all AWS resources provisioned. | `map(string)` | `{}` | no |
```
[the remark this answers is not in the document]
Comment (jsf9k) [reply terraform/README.md]: See commit 87991b5.

File: shared-services/README.md [hunk 3]
```diff
@@ -61,34 +61,55 @@ future changes by simply running `terraform apply
...
+| provisionssmdocument\_policy\_description | The description to associate with the IAM policy that allows sufficient permissions to provision the SSM Document resource in the Shared Services account. | `string` | `"Allows sufficient permissions to provision the SSM Document resource in the Shared Services account."` | no |
+| provisionssmdocument\_policy\_name | The name to assign the IAM policy that allows sufficient permissions to provision the SSM Document resource in the Shared Services account. | `string` | `"ProvisionSSMDocument"` | no |
+| ssmsession\_role\_description | The description to associate with the IAM role (and policy) that allows creation of SSM SessionManager sessions to any EC2 instance in this account. | `string` | `"Allows creation of SSM SessionManager sessions to any EC2 instance in this account."` | no |
+| ssmsession\_role\_name | The name to assign the IAM role (and policy) that allows creation of SSM SessionManager sessions to any EC2 instance in this account. | `string` | `"StartStopSSMSession"` | no |
 | tags | Tags to apply to all AWS resources provisioned. | `map(string)` | `{}` | no |
```
[the remark this answers is not in the document]
Comment (jsf9k) [reply shared-services/README.md]: See commit 87991b5.

File: master/README.md [hunk 4]
```diff
@@ -60,32 +60,46 @@ future changes by simply running `terraform apply
...
+| organizationsreadonly\_role\_description | The description to associate with the IAM role that allows read-only access to all AWS Organizations information in the Master account. | `string` | `"Allows read-only access to all AWS Organizations information in the Master account."` | no |
+| organizationsreadonly\_role\_name | The name to assign the IAM role that allows read-only access to all AWS Organizations information in the Master account. | `string` | `"OrganizationsReadOnly"` | no |
+| provisionaccount\_role\_description | The description to associate with the IAM role that allows sufficient permissions to provision all AWS resources in the Master account. | `string` | `"Allows sufficient permissions to provision all AWS resources in the Master account."` | no |
+| provisionaccount\_role\_name | The name to assign the IAM role that allows sufficient permissions to provision all AWS resources in the Master account. | `string` | `"ProvisionAccount"` | no |
 | tags | Tags to apply to all AWS resources provisioned. | `map(string)` | `{}` | no |
```
[the remark this answers is not in the document]
Comment (jsf9k) [reply master/README.md]: See commit 87991b5.

File: log-archive/README.md [hunk 5]
```diff
@@ -61,27 +61,37 @@ future changes by simply running `terraform apply
...
-| provisionaccount_role_name | The name to assign the IAM role that allows sufficient permissions to provision all AWS resources in the Log Archive account. | `string` | `ProvisionAccount` | no |
+| aws\_region | The AWS region where the non-global resources for the Log Archive account are to be provisioned (e.g. "us-east-1"). | `string` | `"us-east-1"` | no |
+| provisionaccount\_role\_description | The description to associate with the IAM role that allows sufficient permissions to provision all AWS resources in the Log Archive account. | `string` | `"Allows sufficient permissions to provision all AWS resources in the Log Archive account."` | no |
+| provisionaccount\_role\_name | The name to assign the IAM role that allows sufficient permissions to provision all AWS resources in the Log Archive account. | `string` | `"ProvisionAccount"` | no |
 | tags | Tags to apply to all AWS resources provisioned. | `map(string)` | `{}` | no |
```
[the remark this answers is not in the document]
Comment (jsf9k) [reply log-archive/README.md]: See commit 87991b5.

File: images/README.md [hunk 6]
```diff
@@ -65,54 +65,107 @@ future changes by simply running `terraform apply
...
+| provisionthirdpartybucketreadroles\_role\_name | The name to assign the IAM role (as well as the corresponding policy) with the ability to create IAM roles that can read objects in the third-party file storage S3 bucket in the Images account. | `string` | `"ProvisionThirdPartyBucketReadRoles"` | no |
+| provisionvpcs\_policy\_description | The description to associate with the IAM policy that allows sufficient permissions to provision VPCs (and related resources) in the Images account. | `string` | `"Allows sufficient permissions to provision VPCs (and related resources) in the Images account."` | no |
+| provisionvpcs\_policy\_name | The name to assign the IAM policy that allows sufficient permissions to provision VPCs (and related resources) in the Images account. | `string` | `"ProvisionVPCs"` | no |
 | tags | Tags to apply to all AWS resources provisioned. | `map(string)` | `{}` | no |
```
[the remark this answers is not in the document]
Comment (jsf9k) [reply images/README.md]: See commit 87991b5.

File: dynamic/README.md [hunk 7]
```diff
@@ -62,28 +62,38 @@ future changes by simply running `terraform apply
...
-| provisionaccount_role_name | The name to assign the IAM role that allows sufficient permissions to provision all AWS resources in the dynamic account. | `string` | `ProvisionAccount` | no |
+| aws\_region | The AWS region where the non-global resources for the dynamic account are to be provisioned (e.g. "us-east-1"). | `string` | `"us-east-1"` | no |
+| dynamic\_account\_name | The name of the dynamic account to be provisioned. | `string` | n/a | yes |
+| provisionaccount\_role\_description | The description to associate with the IAM role that allows sufficient permissions to provision all AWS resources in the dynamic account. | `string` | `"Allows sufficient permissions to provision all AWS resources in the dynamic account."` | no |
+| provisionaccount\_role\_name | The name to assign the IAM role that allows sufficient permissions to provision all AWS resources in the dynamic account. | `string` | `"ProvisionAccount"` | no |
 | tags | Tags to apply to all AWS resources provisioned. | `map(string)` | `{}` | no |
```
[the remark this answers is not in the document]
Comment (jsf9k) [reply dynamic/README.md]: See commit 87991b5.

File: dns/README.md [hunk 8]
```diff
@@ -60,32 +60,47 @@ future changes by simply running `terraform apply
...
+| provisionaccount\_role\_description | The description to associate with the IAM role that allows sufficient permissions to provision all AWS resources in the DNS account. | `string` | `"Allows sufficient permissions to provision all AWS resources in the DNS account."` | no |
+| provisionaccount\_role\_name | The name to assign the IAM role that allows sufficient permissions to provision all AWS resources in the DNS account. | `string` | `"ProvisionAccount"` | no |
+| provisionroute53\_role\_description | The description to associate with the IAM role (as well as the corresponding policy) that allows sufficient permissions to provision Route 53 in the DNS account. | `string` | `"Allows sufficient permissions to provision Route 53 in the DNS account."` | no |
+| provisionroute53\_role\_name | The name to assign the IAM role (as well as the corresponding policy) that allows sufficient permissions to provision Route 53 in the DNS account. | `string` | `"ProvisionRoute53"` | no |
 | tags | Tags to apply to all AWS resources provisioned. | `map(string)` | `{}` | no |
```
[the remark this answers is not in the document]
Comment (jsf9k) [reply dns/README.md]: See commit 87991b5.

File: audit/README.md [hunk 9]
```diff
@@ -60,27 +60,37 @@ future changes by simply running `terraform apply
...
-| provisionaccount_role_description | The description to associate with the IAM role that allows sufficient permissions to provision all AWS resources in the Audit account. | `string` | `Allows sufficient permissions to provision all AWS resources in the Audit account.` | no |
-| provisionaccount_role_name | The name to assign the IAM role that allows sufficient permissions to provision all AWS resources in the Audit account. | `string` | `ProvisionAccount` | no |
+| aws\_region | The AWS region where the non-global resources for the Audit account are to be provisioned (e.g. "us-east-1"). | `string` | `"us-east-1"` | no |
+| provisionaccount\_role\_description | The description to associate with the IAM role that allows sufficient permissions to provision all AWS resources in the Audit account. | `string` | `"Allows sufficient permissions to provision all AWS resources in the Audit account."` | no |
+| provisionaccount\_role\_name | The name to assign the IAM role that allows sufficient permissions to provision all AWS resources in the Audit account. | `string` | `"ProvisionAccount"` | no |
 | tags | Tags to apply to all AWS resources provisioned. | `map(string)` | `{}` | no |
```
[the remark this answers is not in the document]
Comment (jsf9k) [reply audit/README.md]: See commit 87991b5.

File: users/README.md [hunk 10]
```diff
@@ -65,41 +65,64 @@ future changes by simply running `terraform apply
...
+| assume\_any\_role\_anywhere\_policy\_description | The description to associate with the IAM policy that allows assumption of any role in any account, so long as it has a trust relationship with the Users account. | `string` | `"Allow assumption of any role in any account, so long as it has a trust relationship with the Users account."` | no |
+| assume\_any\_role\_anywhere\_policy\_name | The name to assign the IAM policy that allows assumption of any role in any account, so long as it has a trust relationship with the Users account. | `string` | `"AssumeAnyRoleAnywhere"` | no |
+| aws\_region | The AWS region where the non-global resources for this account are to be provisioned (e.g. "us-east-1"). | `string` | `"us-east-1"` | no |
+| godlike\_usernames | The usernames associated with the god-like accounts to be created, which are allowed to access the terraform backend, are IAM administrators for the Users account, and are allowed to assume any role that has a trust relationship with the Users account.  The format first.last is recommended.  Example: ["firstname1.lastname1",  "firstname2.lastname2"] | `list(string)` | n/a | yes |
```
[the remark this answers is not in the document]
Comment (jsf9k) [reply users/README.md]: See commit fa2a0c3.

**Pull request merged**